### PR TITLE
Release Version 1.0.0

### PR DESCRIPTION
**Pull request by Olaniyan99**
***Purpose***
This PR releases version 1.0.0 of Notification service

***Implementation***
The functionality to get subscribers for a notification was refactored to use the `followers` route in Library service.

CI/CD scripts were added 

Sentry was configured

***Story***
This PR addresses Clubhouse story: https://app.clubhouse.io/clarkcan/story/1636/add-functionality-in-learning-object-service-to-send-amqp-message-on-learning-object-revision-release

### CODE REVIEW

File: internal/pkg/interactors/event-handler.go [hunk 1]
```diff
@@ -45,32 +50,41 @@ const (
...
 	// FIXME: Synchronous HTTP requests should not be used
-	response, err := http.Get(common.GetLibraryURI() + common.FetchSubscribersRoute(author, id))
+	//response, err := http.Get(common.GetLibraryURI() + common.FetchSubscribersRoute(author, cuid, version))
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", common.GetLibraryURI()+common.FetchSubscribersRoute(author, cuid, version), nil)
+	req.Header.Set("Authorization", "Bearer "+signedString)
+	req.Header.Set("Content-Type", "application/json")
+
+	response, err := client.Do(req)
+	log.Print(response.StatusCode)
```
Comment (pzalep1) [inline internal/pkg/interactors/event-handler.go]: Should we be logging these errors to sentry? @nvisal1

File: internal/pkg/interactors/event-handler.go [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
-	Payload *Payload `json:"payload"`
+	Category  int              `json:"category"`
+	Payload   *Payload         `json:"payload"`
+	Requester common.UserToken `json:"requester"`
 }
 
 // Payload represents the embedded information on
 // a service event
 type Payload struct {
-	Username string `json:"username"`
```
Comment (pzalep1) [inline internal/pkg/interactors/event-handler.go]: Is there a reason these are uppercase variable names?

File: internal/pkg/interactors/event-handler.go [hunk 3]
```diff
@@ -45,32 +50,41 @@ const (
 func RegisterEvent(datastore common.DataStore, event ServiceEvent) {
 	switch event.Category {
 	case AuthorUpdatedLearningObject:
-		notifySubscribersOfUpdate(datastore, *event.Payload)
-	break
+		notifySubscribersOfUpdate(datastore, *event.Payload, event.Requester)
+		break
```
Comment (pzalep1) [inline internal/pkg/interactors/event-handler.go]: Do these need to be that far indented?

**Comment by pzalep1** [comment]
This is more of a general comment, but have we decided on if we are going to being referring to the followers of a learning object as followers or subscribers? It might be a good idea to use the same vocab in comments/types throughout all of our services.

File: internal/pkg/interactors/event-handler.go (hunk 1 above)
Comment (nvisal1) [reply internal/pkg/interactors/event-handler.go]: I would like to see Sentry in this service. However, I think that it is out of the scope of this PR.

File: internal/pkg/interactors/event-handler.go (hunk 2 above)
Comment (nvisal1) [reply internal/pkg/interactors/event-handler.go]: Yes. In Golang, uppercase functions and variables can be used outside of the package that it is declared in. 

https://tour.golang.org/basics/3

File: internal/pkg/interactors/event-handler.go (hunk 3 above)
Comment (Olaniyan99) [reply internal/pkg/interactors/event-handler.go]: i didnt notice that, i'll fix that, thank you

**Comment by pzalep1** [comment]
Also, if there are any updates to the API contract or routes can we update the README as well?

File: internal/pkg/interactors/event-handler.go (hunk 3 above)
Comment (Olaniyan99) [reply internal/pkg/interactors/event-handler.go]: These indents are actually not present in the code, not sure why they show up here like that.

File: internal/pkg/common/config.go [hunk 4]
```diff
@@ -3,13 +3,14 @@ package common
...
+	"strconv"
 )
 
 // FetchSubscribersRoute is an exported function. The function
 // accepts a two string values (username, id) and returns a string
 // value of the route to get Learning Object subscribers.
-func FetchSubscribersRoute(username string, id string) string {
-	return "/learning-objects/" + username + "/" + id + "/subscribers"
+func FetchSubscribersRoute(username string, cuid string, version int) string {
+	return "/users/" + username + "/learning-objects/" +cuid+ "/version/" +strconv.Itoa(version)+ "/followers"
```
Comment (nvisal1) [inline internal/pkg/common/config.go]:
```suggestion
	return "/users/" + username + "/learning-objects/" + cuid + "/version/" + strconv.Itoa(version) + "/followers"
```

**Pull request merged**